frontend/src/metabase/components/PasswordReveal.jsx

```diff
@@ -62,6 +62,7 @@ export default class PasswordReveal extends Component {
 
         <div className="ml-auto flex align-center">
           <a
+            data-testid="visibility-toggle"
```

Comment:
I share that view. We've mostly avoided testing ids across Cypress and unit tests. In addition to not resembling how the software is used, it litters your app code with test-specific labels, which feels messy. 

Using text/label/placeholders gets us 90% of the way there. Our Cypress tests frequently use `.Icon-[NAME]` classes, which I think covers most of the rest. I wouldn't mind using testing ids over css class for icons, since there's only one component that would need to be updated. Implementing that with an accessibility-friendly selection method sounds great too.

I think inevitably, there are some cases where using fragile css selectors or testing ids is the only way. For example, we have some tests that grab `<rect>` elements out of our visualizations which are created by a third-party library. In cases like that, I don't think we should be too dogmatic. IMO it's more about limiting the scope of fragile tests or encapsulating the fragility rather than jumping through hoops to totally ban them.

Reply:
> I don't think we should be too dogmatic. IMO it's more about limiting the scope of fragile tests or encapsulating the fragility rather than jumping through hoops to totally ban them.

+1

I can see how improving overall accessibility is a good thing (and I appreciate that RTL was intentionally designed to encourage it), but that's not a small feat in a code base of this size. It would require a concerted effort. In meantime, I can try to include more queries using [implicit roles](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time) (where possible).
If icons are not likely to change, and (as you said) they impact only one file - that gives us a more control + yields huge benefits. If I get stuck with some other elements in future tests, I'll consult with you regarding the appropriate selector.

Thank you for clarifying this.